pkg/config/configuration_test.go

```diff
@@ -479,29 +492,69 @@ func TestSetTracingSpecFromEnv(t *testing.T) {
 	t.Setenv("OTEL_EXPORTER_OTLP_ENDPOINT", "http://otlpendpoint:1234")
 	t.Setenv("OTEL_EXPORTER_OTLP_INSECURE", "true")
 	t.Setenv("OTEL_EXPORTER_OTLP_PROTOCOL", "http/json")
+	t.Setenv("OTEL_EXPORTER_OTLP_HEADERS", "api-key1=value1,api-key2=value2")
 
 	// get default configuration
 	conf := LoadDefaultConfiguration()
 
 	// set tracing spec from env
-	SetTracingSpecFromEnv(conf)
+	err := SetTracingSpecFromEnv(conf)
+	require.NoError(t, err)
 
 	assert.Equal(t, "otlpendpoint:1234", conf.Spec.TracingSpec.Otel.EndpointAddress)
 	assert.Equal(t, "http", conf.Spec.TracingSpec.Otel.Protocol)
 	require.False(t, conf.Spec.TracingSpec.Otel.GetIsSecure())
+	assert.Equal(t, "api-key1=value1,api-key2=value2", conf.Spec.TracingSpec.Otel.Headers)
 
 	// Spec from config file should not be overridden
 	conf = LoadDefaultConfiguration()
 	conf.Spec.TracingSpec.Otel.EndpointAddress = "configfileendpoint:4321"
 	conf.Spec.TracingSpec.Otel.Protocol = "grpc"
 	conf.Spec.TracingSpec.Otel.IsSecure = ptr.Of(true)
+	conf.Spec.TracingSpec.Otel.Headers = "another-key1=value1,another-key2=value2"
 
 	// set tracing spec from env
-	SetTracingSpecFromEnv(conf)
+	err = SetTracingSpecFromEnv(conf)
+	require.NoError(t, err)
 
 	assert.Equal(t, "configfileendpoint:4321", conf.Spec.TracingSpec.Otel.EndpointAddress)
 	assert.Equal(t, "grpc", conf.Spec.TracingSpec.Otel.Protocol)
 	require.True(t, conf.Spec.TracingSpec.Otel.GetIsSecure())
+	assert.Equal(t, "another-key1=value1,another-key2=value2", conf.Spec.TracingSpec.Otel.Headers)
+}
+
+func TestTracingPrecedenceFromEnv(t *testing.T) {
+	t.Setenv(env.OtlpExporterTracesEndpoint, "tracesendpoint:4321")
+	t.Setenv(env.OtlpExporterEndpoint, "configfileendpoint:4321")
+	t.Setenv(env.OtlpExporterTracesProtocol, "grpc")
+	t.Setenv(env.OtlpExporterProtocol, "http")
+	t.Setenv(env.OtlpExporterTracesHeaders, "traces-key1=value1,traces-key2=value2")
+	t.Setenv(env.OtlpExporterHeaders, "key1=value1,key2=value2")
+	t.Setenv(env.OtlpExporterTracesTimeout, "2000")
+	t.Setenv(env.OtlpExporterTimeout, "1000")
+
+	conf := LoadDefaultConfiguration()
+	err := SetTracingSpecFromEnv(conf)
+	require.NoError(t, err)
+
+	assert.Equal(t, "tracesendpoint:4321", conf.Spec.TracingSpec.Otel.EndpointAddress)
+	assert.Equal(t, "grpc", conf.Spec.TracingSpec.Otel.Protocol)
+	assert.Equal(t, "traces-key1=value1,traces-key2=value2", conf.Spec.TracingSpec.Otel.Headers)
+	assert.Equal(t, 2000, conf.Spec.TracingSpec.Otel.Timeout)
+}
+
+func TestTracingTimeoutFromEnv(t *testing.T) {
+	t.Setenv(env.OtlpExporterTracesTimeout, "invalid")
+	conf := LoadDefaultConfiguration()
+	err := SetTracingSpecFromEnv(conf)
+	require.NoError(t, err)
```

Comment:
```suggestion
	require.Error(t, err)
```
This test case looks like it should always error with an invalid string to int conversion?

Reply:
Yes, correct, fixed. Thanks.